### DISCUSSION TITLE
Ability to backup metrics

### DISCUSSION THREAD

**Discussion by stelfrag** [netdata/netdata#12229] [Ideas]
Starting this discussion inspired by issue https://github.com/netdata/netdata/issues/11849  by [ElasMiratoris](https://github.com/ElasMiratoris)

> Having a method to backup & restore Netdata metrics (including child nodes) would be much appreciated.

Idea

- A "snapshot" of the database (metrics and associated metadata required for later exploration) for a certain time period
- Ability to view that data using an agent perhaps in "view only mode" 
  - In the linked issue this was "simulated" by activating a new container with restored data in order to access the dashboard
    - The new agent attempts to "cleanup the database" (and of course collect new metrics)
    - May not actually need to actively collect metrics (the main function as I understand it is to view past data)
    - Needs associated metadata to correctly identify 
       - Stored metrics (charts / dimensions)
       - Any children that have streamed data and those have been stored in the "snapshot"

**Comment by ElasMiratoris** [comment]
`May not actually need to actively collect metrics (the main function as I understand it is to view past data)`  
This is correct. I am only looking for a way to automatically archive and view historical data.  

Netdata currently has a snapshot feature, that comes very close to what I need. Unfortunately the snapshots are assembled in the Javascript frontend. As such they can not be retrieved via the API.  
While it would be possible to work around this with an autoclicker, I am looking for a solution, that is more elegant and stable.

**Reply by andrewm4894** [reply]
I'd love some more modern snapshot formats like apache parquet files or something a bit more portable too.

**Reply by ElasMiratoris** [reply]
Apache parquet is certainly interesting. While I don't have the ambition to ingest the data into Hadoop, having that portability would be handy.
That being said, my primary use case is to be able to look at historical data in Netdata itself.
How much implementation work do you think it would be to implement parquet?
Are there libraries that make it easy?

**Reply by andrewm4894** [reply]
oh no idea myself im afraid (i am the ml guy and dont know c/c++). But i do see these. 

- https://github.com/apache/arrow/tree/master/c_glib
- https://github.com/apache/arrow/tree/master/cpp

But i'd imagine implementing them would be a bigger sort of project.

My angle on this would be for something like a `api/v1/dataframe` api or way to just get parquet files from the agent to maybe enable some ML stuff once we can get a good chunk of data easily out of the agent. So i think i'm side tracking this discussion a little potentially :)

I think netdata cloud based storage of such snapshots could be a useful nice feature for some users perhaps too. In addition to maybe being able to store them in agent too if some users would prefer that too.
